python/ipywidgets/ipywidgets/embed.py

```diff
@@ -297,9 +310,44 @@ def embed_minimal_html(fp, views, title='IPyWidget export', template=None, **kwa
         This should be a Python string with placeholders
         `{{title}}` and `{{snippet}}`. The `{{snippet}}` placeholder
         will be replaced by all the widgets.
-    {embed_kwargs}
+    drop_defaults: boolean
+        Whether to drop default values from the widget states.
+    state: dict, string or None (default)
+        The state to include. When set to None or "dependent", the state of 
+        all passed views is included. When set to "complete", the complete
+        state of the widget is included. Otherwise it uses the passed state 
+        directly. This allows for end users to include a smaller state, under 
+        the responsibility that this state is sufficient to reconstruct the 
+        embedded views.
+    indent: integer, string or None
+        The indent to use for the JSON state dump. See `json.dumps` for
+        full description.
+    embed_url: string or None
+        Allows for overriding the URL used to fetch the widget manager
+        for the embedded code. This defaults (None) to a `jsDelivr` CDN url.
+    requirejs: boolean (True)
+        Enables the requirejs-based embedding, which allows for custom widgets.
+        If True, the embed_url should point to an AMD module.
+    cors: boolean (True)
+        If True avoids sending user credentials while requesting the scripts.
+        When opening an HTML file from disk, some browsers may refuse to load
+        the scripts.
     """
-    snippet = embed_snippet(views, **kwargs)
+
+    if state is None or state == "dependent":
```

Comment:
Changing the default (`None` case) is the most controversial here, as it is strictly speaking backwards incompatible. There might be a case for calling it a workaround for the poor GC of ipywidgets, but making this change should at least have some more eyes on it before merging.

Reply:
I understand your argument and it's difficult for me to estimate how much of an impact this change would have to other projects. Coming from [ipyleaflet](https://github.com/jupyter-widgets/ipyleaflet/issues/1098), it just looks like a bug to me and then I'd argue, that backwards compatibility should not rely on bugs. However, if other people really need the whole state and really have that implementation assumption as well, then I agree, we should stay backwards compatible as well. What do you think?

Your [solution](https://github.com/jupyter-widgets/ipywidgets/issues/3448#issuecomment-1459919374) definelty will work in the end but as a user, it's not so clear to me when I'd use the dependent state and when the complete.

Reply:
I kind of agree it is a bug, we didn't implement this feature for nothing. I think this was the intended behavior, but yeah, fixing a bug can be considered a breaking change :)

Reply:
See original discussion here: https://github.com/jupyter-widgets/ipywidgets/pull/1387#discussion_r119173856